sql/mysqld.cc

```diff
@@ -6237,7 +6237,7 @@ static int generate_server_uuid() {
 
   delete thd;
 
-  strncpy(server_uuid, uuid.c_ptr(), sizeof(server_uuid));
+  strncpy(server_uuid, uuid.c_ptr(), sizeof(server_uuid) - 1);
```

Comment:
It did not complain for me in the ubuntu:focal docker image with gcc 9.4.0. It also have seemed to have complained only on native focal. The warnings in sql/wsrep_high_priority_service.cc and sql/wsrep_mysqld.cc reproduced didn't reproduce locally and they appeared only in focal docker image. 
But anyway, since there's a null character added later, I'll remove this change as well.